From f1522823d20e5562083bec1f673f9b60888d4c84 Mon Sep 17 00:00:00 2001
From: Wendy Liang <wendy.liang@xilinx.com>
Date: Thu, 16 Jan 2020 13:18:43 -0800
Subject: [PATCH 0855/1851] remoteproc: returns message for peek remote kick
 function

commit 4345095741c9124bde625ca503b70df46f6f2df4 from
https://github.com/Xilinx/linux-xlnx.git

Update the peek remote kick API to also return the message
from remote.

Signed-off-by: Wendy Liang <wendy.liang@xilinx.com>
Reviewed-by: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/remoteproc/remoteproc_internal.h | 4 ++--
 drivers/remoteproc/remoteproc_sysfs.c    | 2 +-
 include/linux/remoteproc.h               | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/remoteproc/remoteproc_internal.h b/drivers/remoteproc/remoteproc_internal.h
index b563ef36d8a9..05a7e629fdd9 100644
--- a/drivers/remoteproc/remoteproc_internal.h
+++ b/drivers/remoteproc/remoteproc_internal.h
@@ -126,10 +126,10 @@ bool rproc_allow_sysfs_kick(struct rproc *rproc)
 }
 
 static inline
-bool rproc_peek_remote_kick(struct rproc *rproc)
+bool rproc_peek_remote_kick(struct rproc *rproc, char *buf, size_t *len)
 {
 	if (rproc->ops->peek_remote_kick)
-		return rproc->ops->peek_remote_kick(rproc);
+		return rproc->ops->peek_remote_kick(rproc, buf, len);
 	else
 		return false;
 }
diff --git a/drivers/remoteproc/remoteproc_sysfs.c b/drivers/remoteproc/remoteproc_sysfs.c
index e680d32ba89f..596203c4ac79 100644
--- a/drivers/remoteproc/remoteproc_sysfs.c
+++ b/drivers/remoteproc/remoteproc_sysfs.c
@@ -164,7 +164,7 @@ static ssize_t remote_kick_show(struct device *dev,
 
 	buf[0] = '0';
 	buf[1] = '\n';
-	if (rproc_peek_remote_kick(rproc))
+	if (rproc_peek_remote_kick(rproc, NULL, NULL))
 		buf[0] = '1';
 	return 2;
 }
diff --git a/include/linux/remoteproc.h b/include/linux/remoteproc.h
index a2758fce3602..56fa63d27e95 100644
--- a/include/linux/remoteproc.h
+++ b/include/linux/remoteproc.h
@@ -376,7 +376,7 @@ struct rproc_ops {
 	int (*start)(struct rproc *rproc);
 	int (*stop)(struct rproc *rproc);
 	void (*kick)(struct rproc *rproc, int vqid);
-	bool (*peek_remote_kick)(struct rproc *rproc);
+	bool (*peek_remote_kick)(struct rproc *rproc, char *buf, size_t *len);
 	void (*ack_remote_kick)(struct rproc *rproc);
 	void * (*da_to_va)(struct rproc *rproc, u64 da, int len);
 	int (*parse_fw)(struct rproc *rproc, const struct firmware *fw);
-- 
2.31.1

